cluster/juju/layers/kubernetes-master/tactics/update_addons.py

```diff
@@ -54,32 +60,40 @@ def add_addon(source, dest):
     with open(dest, "w") as f:
         f.write(content)
 
+
 def update_addons(dest):
     """ Update addons. This will clean the addons folder and add new manifests
     from upstream. """
     clean_addon_dir(dest)
     with kubernetes_repo() as repo:
-        add_addon(repo + "/cluster/addons/dashboard/dashboard-controller.yaml", dest)
-        add_addon(repo + "/cluster/addons/dashboard/dashboard-service.yaml", dest)
-        add_addon(repo + "/cluster/addons/dns/kubedns-controller.yaml.in", dest + "/kubedns-controller.yaml")
-        add_addon(repo + "/cluster/addons/dns/kubedns-svc.yaml.in", dest + "/kubedns-svc.yaml")
-        add_addon(repo + "/cluster/addons/cluster-monitoring/influxdb/grafana-service.yaml", dest)
-        add_addon(repo + "/cluster/addons/cluster-monitoring/influxdb/heapster-controller.yaml", dest)
-        add_addon(repo + "/cluster/addons/cluster-monitoring/influxdb/heapster-service.yaml", dest)
-        add_addon(repo + "/cluster/addons/cluster-monitoring/influxdb/influxdb-grafana-controller.yaml", dest)
-        add_addon(repo + "/cluster/addons/cluster-monitoring/influxdb/influxdb-service.yaml", dest)
+        add_addon(repo + "/cluster/addons/dashboard/dashboard-controller.yaml",
+                  dest)
+        add_addon(repo + "/cluster/addons/dashboard/dashboard-service.yaml",
+                  dest)
+        add_addon(repo + "/cluster/addons/dns/kubedns-controller.yaml.in",
+                  dest + "/kubedns-controller.yaml")
+        add_addon(repo + "/cluster/addons/dns/kubedns-svc.yaml.in",
+                  dest + "/kubedns-svc.yaml")
+        influxdb = "/cluster/addons/cluster-monitoring/influxdb"
+        add_addon(repo + influxdb + "/grafana-service.yaml", dest)
+        add_addon(repo + influxdb + "/heapster-controller.yaml", dest)
+        add_addon(repo + influxdb + "/heapster-service.yaml", dest)
+        add_addon(repo + influxdb + "/influxdb-grafana-controller.yaml", dest)
+        add_addon(repo + influxdb + "/influxdb-service.yaml", dest)
 
 # Entry points
 
+
 class UpdateAddonsTactic(Tactic):
     """ This tactic is used by charm-tools to dynamically populate the
     template/addons folder at `charm build` time. """
 
     @classmethod
-    def trigger(cls, relpath):
+    def trigger(cls, entity, target, layer, next_config):
         """ Determines which files the tactic should apply to. We only want
         this tactic to trigger once, so let's use the templates/ folder
         """
+        relpath = entity.relpath(layer.directory) if layer else entity
```

Comment:
It should be noted that a fix did go in to charm-tools that made the signature change work as expected, but @johnsca told me this would fix problems with older charm-tools that we were getting in the charmbox containers in our ci process.